msal/application.py

```diff
@@ -160,6 +159,39 @@ class ClientApplication(object):
 
     ATTEMPT_REGION_DISCOVERY = True  # "TryAutoDetect"
 
+    _known_authorities = None
+
+    @classmethod
+    def set_known_authorities(cls, known_authorities):
+        """Declare a list of domain names which you allow MSAL to operate with.
+
+        If your app operates with some authorities that you know and own,
+        such as some ADFS or B2C or private cloud,
+        it is recommended and sometimes required that you declare them here,
+        so that MSAL will use them without any validation.
+
+        This setting shall be configured at most once, during your app start-up,
+        rather than being changed to different authorities during runtime.
+
+        :param list[str] known_authorities:
+            Authorities that you known, for example::
+
+                [
+                    "contoso.com",  # Your own domain
```

Comment:
The variable name is "known_authorities" but these are not authorities, these are environments. I think environments mae sense, so is the var name wrong?

Reply:
Excellent point! I was also torn between different names.

The original Known Authorities specs chose the name, and its pseudo code sometimes contains a full url, sometimes a host name only. And that ambiguity was also observed in MSAL Js, where it was [documented as URI](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/configuration.md#auth-config-options), but the [inline comment suggested otherwise](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/v1.3.4/lib/msal-node/src/config/Configuration.ts#L17). CC @sameerag 

Now I inclined to rename this to `known_hosts`. Thoughts?

Reply:
known_authority_hosts :)

Reply:
Renamed to `known_authority_hosts` in the latest commit.